pos_require_product_quantity/static/src/js/pos.js

```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright 2019 Coop IT Easy SCRLfs
+    Robin Keunen <robin@coopiteasy.be>
+  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+*/
+
+odoo.define(
+
+    'pos_require_product_quantity.pos_require_product_quantity',
+    function (require) {
+        "use strict";
+
+        var core = require('web.core');
+        var screens = require("point_of_sale.screens");
+
+        var _t = core._t;
+
+        screens.ActionpadWidget = screens.ActionpadWidget.include({
+            renderElement: function () {
+                var self = this;
+                this._super();
+
+                this.$('.pay').click(function () {
+
+                    if (self.pos.config.require_product_quantity) {
```

Comment:
Two things here:

1) I guess the idea is to prevent the cashier from getting to the payment screen, too.. not just showing the popup.
For this I recommend doing it like this:

https://github.com/OCA/pos/blob/8eca935e1a9fe17976071d7ccf2084eac8654cfc/pos_event_sale/static/src/js/screens.js#L41-L75

This way you respect the original event handlers, which may have been altered by other modules.



2) It's better to avoid messing with the event callbacks if the setting is disabled, than evaluating in the callback.

pseudo-code:

```js
            renderElement: function () {
                if (!this.pos.config.require_product_quantity) {
                    return this._super.apply(this, arguments);
                }

                // your custom code after this..
            }
```